resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+
+    1. Deb lists the resources types he has in his environment
+
+        ```bash
+        rad resource-provider list -e myenv
+        ```
+        Radius lists the custom resource types that Deb has registered in his environment
+
+2. Amy is a system integrator who is familiar with OpenAPI schema format and wants to author a custom resource PostgreSQL in Radius
+
+    1. Amy uses the radCLI to scaffold a yaml template definition
+
+        ```bash
+        rad resource-provider init plaid --template-kind yaml
+        ```
+        A sample yaml file is created in the application folder.
+
+    1. Amy authors the schema for plaid in typespec format
+        
+        ```
+        namespace: 'Applications.Datastores'
+        apiVersion: 'v1'
+        scope: '/planes/radius/Applications.Datastores'
+        resourceTypes:
+            'Applications.Datastores/postgreSQL':
+                name: 'Name of the resource'
+                description: 'PostgreSQL' 
+                properties: 
+                required: ['host', 'env', 'secrets']
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+
+    1. Amy adds a version for the schema
+
+         ```
+        namespace: 'Applications.Datastores'
+        apiVersion: 'v1'
+        scope: '/planes/radius/Applications.Datastores'
+        resourceTypes:
+            'Applications.Datastores/postgreSQL':
+                name: 'Name of the resource'
+                description: 'PostgreSQL' 
+                properties: 
+                required: ['host', 'env', 'secrets']
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+   1. Amy registers the postgreSQL schema to an existing provider in Radius
+    
+        ```bash
+        rad resource-type add -rp Applications --template-kind yaml -e myenv -v v1
+        ```
+        
+        Radius complies the schema and registers the custom resource type Plaid in UCP
+        
+        Question : Will we continue to accept new types like postgreSQL to `Applications` namespace
+   
```

Comment:
BTW I feel like this is a great way place to talk about community supported assets like a "contrib" repository of resource types. Now that we've explained the work to define one we can talk about the goal of reuse with more clarity.